WizFi310/WizFi310.cpp

```diff
@@ -305,19 +305,13 @@ void WizFi310::_packet_handler()
         return;
     }
 
-    struct packet *packet = new struct packet;
+    struct packet *packet = new struct packet(id, amount);
     if (!packet) {
         return;
     }
 
-    packet->id = id;
-    packet->len = amount;
-    packet->next = 0;
-    packet->data = (char*)malloc(amount);
-
-    
     if (!(_parser.read((char*)packet->data, amount))) {
-        free(packet);
+        delete(packet);
```

Comment:
Is this a use of the delete operator ?
It might be a bit misleading to see it written like that instead of `delete packet`.

Reply:
Yes it is. I've seen delete used both ways. Feel free to change it if you prefer the other way